tests/Microsoft.Identity.Test.Unit.net45/PublicApiTests/ClientCredentialWithCertTest.cs

```diff
@@ -139,6 +159,49 @@ public async Task JsonWebTokenWithX509PublicCertSendCertificateOnBehalfOfTestAsy
             }
         }
 
+        [TestMethod]
+        [Description("Test for client assertion with X509 public certificate using sendCertificate")]
+        public async Task JsonWebTokenWithX509PublicCertSendCertificateSilentTestAsync()
+        {
+            using (var harness = CreateTestHarness())
+            {
+                SetupMocks(harness.HttpManager, "https://login.microsoftonline.com/my-utid/");
+                var certificate = new X509Certificate2(
+                    ResourceHelper.GetTestResourceRelativePath("valid_cert.pfx"),
+                    MsalTestConstants.DefaultPassword);
+
+                var app = ConfidentialClientApplicationBuilder.Create(MsalTestConstants.ClientId)
+                                                              .WithAuthority(
+                                                                  new System.Uri("https://login.microsoftonline.com/my-utid"),
+                                                                  true).WithRedirectUri(MsalTestConstants.RedirectUri)
+                                                              .WithHttpManager(harness.HttpManager)
+                                                              .WithCertificate(certificate).BuildConcrete();
+
+
+
+                _tokenCacheHelper.PopulateCache(app.UserTokenCacheInternal.Accessor);
+                app.UserTokenCacheInternal.Accessor.DeleteAccessToken(
+                    new MsalAccessTokenCacheKey(
+                        MsalTestConstants.ProductionPrefNetworkEnvironment,
+                        MsalTestConstants.Utid,
+                        MsalTestConstants.UserIdentifier,
+                        MsalTestConstants.ClientId,
+                        MsalTestConstants.ScopeForAnotherResourceStr));
+
+                harness.HttpManager.AddMockHandler(CreateTokenResponseHttpHandlerWithX5CValidation(false));
```

Comment:
The mock handler does not seem to make any validation related to the content of that header, just that it is present. Since you are using a known certificate, the value should just be a constant ?